src/app/auth/login/login.component.ts

```diff
@@ -1,30 +1,38 @@
 import { Component } from '@angular/core';
 import { NavigationExtras, Router } from '@angular/router';
 import { AuthService } from '../auth.service';
-import { FormControl, Validators } from '@angular/forms';
+import {FormControl, FormGroup, Validators} from '@angular/forms';
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
-  styleUrls: ['./login.component.css']
+  styleUrls: ['../../shared/_styles/auth.css']
 })
 export class LoginComponent {
+  form = new FormGroup({
+    username: new FormControl(),
+    password: new FormControl()
+  });
+
   message: string;
   hide = true;
   email = new FormControl('', [Validators.required, Validators.email]);
   constructor(public authService: AuthService, public router: Router) {
-    this.setMessage();
   }
 
-  setMessage() {
-    this.message = 'Logged ' + (this.authService.isLoggedIn ? 'in' : 'out');
+  setMessage() { // TODO This needs to work somehow....
+    if (this.authService.isLoginFailed === true){
```

Comment:
If you limit yourself to what you were taught or already know, you'll criple yourself (and you know, since you are learning security on your own). Always do the best you can and never accept the lack of knowledge as a barrier, take it as a challenge instead.